2014_TPDS_basic_decomposition_based_algorithm/src/handle_segment.rs

```diff
@@ -50,13 +57,55 @@ pub fn create_segments(dag: &mut Graph<NodeData, i32>) -> Vec<Segment> {
                 && segment.end_range <= node.params["earliest_finish_time"]
             {
                 segment.nodes.push(node.clone());
+                segment.parallel_degree += 1;
             }
         }
     }
 
     segments
 }
 
+fn classify_segment(volume: f32, period: f32, crit_path_len: f32, segment: &mut Segment) {
+    if segment.nodes.is_empty() {
+        unreachable!("Segment is empty")
+    }
+
+    segment.classification =
+        if segment.parallel_degree as f32 > volume / ((2.0 * period) - crit_path_len) {
+            Some(SegmentClassification::Heavy)
+        } else {
+            Some(SegmentClassification::Light)
+        };
+}
+
+#[allow(dead_code)] //TODO: remove
+fn classify_segments(
```

Comment:
```suggestion
fn classify_dag(
```

Reply:
I forgot to change it. Thank you very much.